### ISSUE TITLE
DL3042 false positive, `--no-chache-dir` is a few lines below

### ISSUE THREAD

**Issue by NickLarsenNZ** [hadolint/hadolint#1042]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

There should be no lint error

### Actual behavior

```
Dockerfile:5 DL3042 warning: Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.8.1@sha256:e87caa74dcb7d46cd820352bfea12591f3dba3ddc4285e19c7dcd13359f7cefd

FROM python:3.12

RUN python3 -m venv /app \
    && source /app/bin/activate \
    && pip install \
        --no-cache-dir \
        --upgrade \
        tzdata
```

**Additional environment details (OS, stack version, etc.)**

- NixOS
- Github Actions runner (ubuntu-latest)

**Comment by m-ildefons** [comment]
```
│ ~ │► hadolint --version
Haskell Dockerfile Linter 2.13.1
│ ~ │► cat tmp/Dockerfile
# syntax=docker/dockerfile:1.8.1@sha256:e87caa74dcb7d46cd820352bfea12591f3dba3ddc4285e19c7dcd13359f7cefd

FROM python:3.12

RUN python3 -m venv /app \
 && source /app/bin/activate \
 && pip install \
     --no-cache-dir \
     --upgrade \
     tzdata
│ ~ │► hadolint tmp/Dockerfile
tmp/Dockerfile:5 DL3013 warning: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`
tmp/Dockerfile:5 SC3046 warning: In POSIX sh, 'source' in place of '.' is undefined.
```

Can't reproduce with the latest version of Hadolint: v2.13.1. In case you're using Hadolint through the GitHub action, use version @v3.2.0 to get Hadolint v2.13.1.
